## Task:
1. Read file
2. Explore file (EDA) 
3. Statisctical analysis
4. Data cleaning if needed.

# Video_Games_Sales_ Analysis 

***Note: Explain using CRIPS-DM methodology for project implementation***
# 1 Introduction 
### 1.1 What is data
A data is a compilation of plain facts that have been gathered within a defined context. Statistians would refer to it as a set of obsevations contained in variables or columns of varying/unique characteristics. Data can also be referred to as a piece of information after it has been summarised, and when subjected to analysis, data can be referred to as evidence of a hypothesis. Before data can become information and utilmately evidence it must go through a process. Data analysis is the process in which raw data is ordered and organized, to be used in methods that help to explain the past and predict the future (Hector, 2013). 

While data analytics as a science does not have a predefined methodolgy as each dataset and context is diffrent, generally, the process of turning raw data into actionable insight can be expected to include early data analysis (EDA), descriptive analysis, statistical analysis, diagonistic analysis, predictive analysis and prescriptive analysis. With all this in mind, regadless of the domain, there is always an ultimate goal that most analytical project seek to achieve from a dataset, which is to derive actionable insight and communicate those insight to help decision makers such as statekholders make informed decision about the context in which the data had been gathered. Most of the time this decision makers are not data experts. While as data professionals, we should be very informed and intrested in the subtle details of analytical methodology, there is a clear gap between the infomation we seek to communicate and the ability of those that need these informations to understand them, most of the time due to lack technical background to understand these methodologies.  This is where Data visualisation come into play. It is logically safe to assume that putting trends and insight in pictural form will be more informative than telling numbers or texts. At the end of the day, a good data analytics implementation should provide a clear picture of where an organisation for example is, where they have been and predict where they are heading to (UCDPA,2024), data visualisation is a tool that can be used to achieve this. 


### 1.2 Data Visualisation
Data visualisation is the graphical representation of raw or processed data (Ravikiran, 2023). It is the brach of data-analytics that involve using visual elements like charts, graphs, and maps, to enhance the understanding of trends and pattern in data (CCT college, 2024).

The use of data visualisation is not limited to commnicating insight only, it is also useful for data cleaning: for example when idenfying out of trend data, exploration of data structures, outlier detection: for example visually identfy data that are out of range of the remain overall observation, clustering and pattern discovery, evaluation of modeling results, and presenting findings (Unwin, 2020).
#### [Data vis not over] Remaing: 
1. Importance of DV to overall DA goal
2. DV in python, breifly explain py viz liabraries
3. Introduce project breifly and whe DV come in play here 

### 1.3 Problem Definition
The domain of this project is a retail company and the context is to analyse data collected on the sales of video games. Comapny wants an analysis that help to determine the sales strategy for the company in their upcoming Winter season based on insight found in their sales record, collected inform of a comma separated version(csv) data file. All implementation must be in a jupyternotebook. Justifucation for all method must be provide usinf markdown

### 1.4 Objective
1. Read csv file in pandas and explore dataset in it raw form using pandas.
2. Perform EDA and statistical analysis
3. Answer project question using appropriate plot and justify 
4. Rationalize all the decisions made in Jupyter Notebook report for companies CTO.
5. Provide a detail conclusion.

### 1.5 May have Objective
1. Using appropriate machien learning algorithm, perform predictive analysis 

#### ReadIng csv file with pandas and exploring dataset in it raw form.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
#import warnings
#warnings.filterwarnings('ignore')

### Liararies used:
1. Pandas
2. Numpy
3. Matplotlib.pyplot 
4. Seaborn

### Explain why each of this liabraries where chosen

In [2]:
df = pd.read_csv('vgsales.csv')

1. What did you do here
2. why conver to dataframe (enables easy data wranglings, Haulage proj might help!)
3. Why stor in a varable, help: same as 2

In [3]:
df.head()

Name Platform  Year_of_Release         Genre Publisher  \
0                Wii Sports      Wii           2006.0        Sports  Nintendo   
1         Super Mario Bros.      NES           1985.0      Platform  Nintendo   
2            Mario Kart Wii      Wii           2008.0        Racing  Nintendo   
3         Wii Sports Resort      Wii           2009.0        Sports  Nintendo   
4  Pokemon Red/Pokemon Blue       GB           1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  Critic_Score  \
0     41.36     28.96      3.77         8.45         82.53          76.0   
1     29.08      3.58      6.81         0.77         40.24           NaN   
2     15.68     12.76      3.79         3.29         35.52          82.0   
3     15.61     10.93      3.28         2.95         32.77          80.0   
4     11.27      8.89     10.22         1.00         31.37           NaN   

   Critic_Count User_Score  User_Count Developer Rating  
0          51.0          8       322.0  Nintendo      E  
1           NaN        NaN         NaN       NaN    NaN  
2          73.0        8.3       709.0  Nintendo      E  
3          73.0          8       192.0  Nintendo      E  
4           NaN        NaN         NaN       NaN    NaN

## EDA

In [4]:
df.shape

(16719, 16)

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16719 entries, 0 to 16718
Data columns (total 16 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16717 non-null  object 
 1   Platform         16719 non-null  object 
 2   Year_of_Release  16450 non-null  float64
 3   Genre            16717 non-null  object 
 4   Publisher        16665 non-null  object 
 5   NA_Sales         16719 non-null  float64
 6   EU_Sales         16719 non-null  float64
 7   JP_Sales         16719 non-null  float64
 8   Other_Sales      16719 non-null  float64
 9   Global_Sales     16719 non-null  float64
 10  Critic_Score     8137 non-null   float64
 11  Critic_Count     8137 non-null   float64
 12  User_Score       10015 non-null  object 
 13  User_Count       7590 non-null   float64
 14  Developer        10096 non-null  object 
 15  Rating           9950 non-null   object 
dtypes: float64(9), object(7)
memory usage: 2.0+ MB


In [6]:
df.isnull().sum()

Name                  2
Platform              0
Year_of_Release     269
Genre                 2
Publisher            54
NA_Sales              0
EU_Sales              0
JP_Sales              0
Other_Sales           0
Global_Sales          0
Critic_Score       8582
Critic_Count       8582
User_Score         6704
User_Count         9129
Developer          6623
Rating             6769
dtype: int64

### Justification: 
***why [head, shape, info]?***
1. The first 3 lines of code where called from pandas liabrary to show the data. In other to ehance data manipulation and wrangling, i transformed the csv data into a dataframe(df) as eairler alluded. Now in other to see the df, the ***[head, shape, ifo]*** methods where used. The head method ***'df.head()'*** will display by default the firs five rows and all the columns in the data. The shape method ***'df.shape'*** was used to show the size of the data, output shows that there are ***16719 observations and 16 columns*** in the data, this not inform me on the size of data i show have in all columns but also gives me an insight into the dimentionality of the data. for instance in highly dimentional data, pandas will display by defaut 20 columns, the shape method will give th actual size if the data it there are more than 20 displayed. The info method ***'df.info()'*** was used for further exploration. The info method was used to get the following informations ***(datatype, shape of actual observation in each variable, name and index of each variable)***. Datatype information helps me to know the structure in which observations in each variable has been stroed. Output shows out of 16 variables, there are 7 categorical/object variables, and 7 numerical variables. This pice of information is important in selceting the appropraite visualsation thechnique to use to display the observations of these variables. ***shape of actual observation in each variable*** information helps me to see if there are non recorded observations(null-values) in any of the observations. Genrally, if there are NAN values, the count of the observation will be less than 16719. The number of NAN will be the diffrence between the actual observation count as shown using the shape method, and the recorded observation count in each variable as shown with the info method. Output show that out of 16 observations, 10 observations namely ***[Name, Year_of_Release,,  Genre, Publisher, Critic_Score, Critic_Count, User_Score, User_Count, Developer, Rating]*** have NAN values in them. This pieces of information is important going forward in terms of data cleaning, relaibility of statistiscal result and even visualisation. To conclude what the 3 codes did was ***showing me the data***.  

***Why [isnull.sum]:***

2. The [isnull.sum] code was used to further explore the missing observations that was found using ***info()*** method. The info method only tells me ther are missing varlues in some observations, implementing isnull.sum method shows me how many are in each of thos variables. ***.sum*** arguments helps to veiw the number of null values in each feuture individially. This information shows the quantity of the missing values, which will guide me on the appropriate approach to handle this problem.

In [7]:
missing = df.isnull().sum()

In [8]:
t_obsevation = len(df)

In [9]:
t_obsevation

16719

In [10]:
df.shape

(16719, 16)

In [11]:
prop = (missing/t_obsevation)*100

In [12]:
prop 

Name                0.011962
Platform            0.000000
Year_of_Release     1.608948
Genre               0.011962
Publisher           0.322986
NA_Sales            0.000000
EU_Sales            0.000000
JP_Sales            0.000000
Other_Sales         0.000000
Global_Sales        0.000000
Critic_Score       51.330821
Critic_Count       51.330821
User_Score         40.098092
User_Count         54.602548
Developer          39.613613
Rating             40.486871
dtype: float64

### Missing observations proportion to actual observations:
***Given the following:***
Proportionality is the quality of corresponding in size or amount to something else(goggle). In this contex, i assumed the observations in each variable is a 100%, out of the 100% observation which is 16719, what persentage out of the 100% is missing values (unrecorded observations).
#### why 'prop = (missing/t_obsevation)*100'

Break down the code: " variable prop is assigned to missing(unrecored observations) divided by total observation(16719) multiplied by 100%."

What this code did is to give me the percentage of all observation that is NAN. This is basically converting the numbers i got using "isnullsum()" into percentage. To do this: 
I assinged "df.isnull().sum()" to variable 'missing' as this: "missing = df.isnull().sum()" and assigned 'len(df)' which will give the count of the total observations in df, to variable "t_observation'. I then used the variables as '(missing/t_obsevation)*100' and assigned the output to variable "prop". I then called prop, which output as shown above is the persentage of the missing values in the columns that have them. 
### Why
This information will further guide me on the most appropriate method to use deal with the missing values.  
# Next
1. Explain the proportions in each variable, and how it impact analysis. 

In [13]:
df["Publisher"].unique()

array(['Nintendo', 'Microsoft Game Studios', 'Take-Two Interactive',
       'Sony Computer Entertainment', 'Activision', 'Ubisoft',
       'Bethesda Softworks', 'Electronic Arts', 'Sega', 'SquareSoft',
       'Atari', '505 Games', 'Capcom', 'GT Interactive',
       'Konami Digital Entertainment', 'Square Enix',
       'Sony Computer Entertainment Europe', 'Virgin Interactive',
       'LucasArts', 'Warner Bros. Interactive Entertainment',
       'Universal Interactive', 'Eidos Interactive', 'RedOctane',
       'Vivendi Games', 'Enix Corporation', 'Namco Bandai Games',
       'Palcom', 'Hasbro Interactive', 'THQ', 'Fox Interactive',
       'Acclaim Entertainment', 'MTV Games', 'Disney Interactive Studios',
       nan, 'Codemasters', 'Majesco Entertainment', 'Red Orb', 'Level 5',
       'Arena Entertainment', 'Midway Games', 'JVC', 'Deep Silver',
       'NCSoft', '989 Studios', 'UEP Systems', 'Parker Bros.', 'Maxis',
       'Imagic', 'Tecmo Koei', 'ASCII Entertainment', 'Valve Software',


In [14]:
df["User_Score"].unique()

array(['8', nan, '8.3', '8.5', '6.6', '8.4', '8.6', '7.7', '6.3', '7.4',
       '8.2', '9', '7.9', '8.1', '8.7', '7.1', '3.4', '5.3', '4.8', '3.2',
       '8.9', '6.4', '7.8', '7.5', '2.6', '7.2', '9.2', '7', '7.3', '4.3',
       '7.6', '5.7', '5', '9.1', '6.5', 'tbd', '8.8', '6.9', '9.4', '6.8',
       '6.1', '6.7', '5.4', '4', '4.9', '4.5', '9.3', '6.2', '4.2', '6',
       '3.7', '4.1', '5.8', '5.6', '5.5', '4.4', '4.6', '5.9', '3.9',
       '3.1', '2.9', '5.2', '3.3', '4.7', '5.1', '3.5', '2.5', '1.9', '3',
       '2.7', '2.2', '2', '9.5', '2.1', '3.6', '2.8', '1.8', '3.8', '0',
       '1.6', '9.6', '2.4', '1.7', '1.1', '0.3', '1.5', '0.7', '1.2',
       '2.3', '0.5', '1.3', '0.2', '0.6', '1.4', '0.9', '1', '9.7'],
      dtype=object)

In [15]:
df["Rating"].unique()

array(['E', nan, 'M', 'T', 'E10+', 'K-A', 'AO', 'EC', 'RP'], dtype=object)

In [16]:
df["Developer"].unique()

array(['Nintendo', nan, 'Good Science Studio', ..., 'Big Red Software',
       'Atomic Games', 'Interchannel-Holon'], dtype=object)

In [17]:
df["Developer"].value_counts()

Ubisoft                                   204
EA Sports                                 172
EA Canada                                 167
Konami                                    162
Capcom                                    139
                                         ... 
Genki, Kojima Productions                   1
Warner Bros. Interactive Entertainment      1
THQ, Altron                                 1
Netherock Ltd.                              1
Interchannel-Holon                          1
Name: Developer, Length: 1696, dtype: int64

## Statistical summary

In [18]:
df.describe()

Year_of_Release      NA_Sales      EU_Sales      JP_Sales  \
count     16450.000000  16719.000000  16719.000000  16719.000000   
mean       2006.487356      0.263330      0.145025      0.077602   
std           5.878995      0.813514      0.503283      0.308818   
min        1980.000000      0.000000      0.000000      0.000000   
25%        2003.000000      0.000000      0.000000      0.000000   
50%        2007.000000      0.080000      0.020000      0.000000   
75%        2010.000000      0.240000      0.110000      0.040000   
max        2020.000000     41.360000     28.960000     10.220000   

        Other_Sales  Global_Sales  Critic_Score  Critic_Count    User_Count  
count  16719.000000  16719.000000   8137.000000   8137.000000   7590.000000  
mean       0.047332      0.533543     68.967679     26.360821    162.229908  
std        0.186710      1.547935     13.938165     18.980495    561.282326  
min        0.000000      0.010000     13.000000      3.000000      4.000000  
25%        0.000000      0.060000     60.000000     12.000000     10.000000  
50%        0.010000      0.170000     71.000000     21.000000     24.000000  
75%        0.030000      0.470000     79.000000     36.000000     81.000000  
max       10.570000     82.530000     98.000000    113.000000  10665.000000

## Next
1. Explain fivr number summary and why (distribution, skewness and all....)
2. Box plot (visually shows outliers and IQR)... why not other plot? 

In [22]:
df.describe(include= 'object')

Name Platform   Genre        Publisher  \
count                         16717    16719   16717            16665   
unique                        11562       31      12              581   
top     Need for Speed: Most Wanted      PS2  Action  Electronic Arts   
freq                             12     2161    3370             1356   

       User_Score Developer Rating  
count       10015     10096   9950  
unique         96      1696      8  
top           tbd   Ubisoft      E  
freq         2425       204   3991

***why df.describe(include= 'object')***
Nutshell: Describes the non-numerical variables 

## TRy imputing with most frequent values 
## Drop and compare with most frequent in terms of the impact on the data distribution 
## 

In [12]:
x = df.isnull().sum()

